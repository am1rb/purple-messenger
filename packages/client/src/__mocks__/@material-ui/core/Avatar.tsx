import React from "react";
import type { AvatarProps } from "@material-ui/core/Avatar";

function MuiAvatar({ src, alt }: AvatarProps) {
  return <img data-testid="mock-mui-avatar" src={src} alt={alt} />;
}

export default MuiAvatar;
