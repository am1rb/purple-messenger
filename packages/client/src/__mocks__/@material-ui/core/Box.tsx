import React from "react";

export interface BoxProps {
  children: React.ReactNode;
}

const muiBox = jest.fn(({ children }: BoxProps) => (
  <div data-testid="mui-box">{children}</div>
));

export default muiBox;
