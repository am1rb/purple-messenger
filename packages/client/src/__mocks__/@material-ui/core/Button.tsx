import React from "react";

export interface ButtonProps {
  onClick?: () => void;
  children: React.ReactNode;
}

function MuiButton({ onClick, children }: ButtonProps) {
  return (
    <div data-testid="mock-mui-button" onClick={onClick}>
      {children}
    </div>
  );
}

export default MuiButton;
