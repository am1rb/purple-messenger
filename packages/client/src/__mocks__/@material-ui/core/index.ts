export { default as Avatar } from "./Avatar";
export { default as Box } from "./Box";
export { default as Card } from "./Card";
export { default as CardContent } from "./CardContent";
export { default as Typography } from "./Typography";
export { default as Button } from "./Button";
