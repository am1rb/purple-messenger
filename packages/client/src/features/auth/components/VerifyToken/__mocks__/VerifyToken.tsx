import React from "react";

function VerifyToken() {
  return <span data-testid="mock-verify-token" />;
}

export default VerifyToken;
