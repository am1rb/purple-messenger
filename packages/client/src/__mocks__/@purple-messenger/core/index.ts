import * as purpleMessengerCore from "@purple-messenger/core";

const mockedPurpleMessengerCore = jest.genMockFromModule<
  typeof purpleMessengerCore
>("@purple-messenger/core");

function mockedAction(action = () => ({ type: "" })) {
  return jest.fn(action);
}

module.exports = {
  ...mockedPurpleMessengerCore,
  loadConversationList: mockedAction(),
  unloadConversationList: mockedAction(),
  signOut: mockedAction(),
  clearMessageList: mockedAction(),
  seenMessageAck: mockedAction(),
  start: mockedAction(),
  shutdown: mockedAction(),
  submitMessage: mockedAction(),
  startTypingMessage: mockedAction(),
  stopTypingMessage: mockedAction(),
  signIn: mockedAction(),
  setAuthError: mockedAction(),
  verifyToken: mockedAction(),
  connected: mockedAction(purpleMessengerCore.connected),
  disconnected: mockedAction(purpleMessengerCore.disconnected),
};
