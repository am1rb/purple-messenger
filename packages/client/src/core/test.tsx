import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { ConnectedRouter } from "connected-react-router";
import configureStore, { history } from "core/redux/configureStore";
import { State as RootState } from "core/redux/reducers";
import theme from "core/style/theme";
import { ThemeProvider } from "@material-ui/styles";

export function renderWithStore(
  ui: React.ReactElement,
  initStore: Partial<RootState>
) {
  const store = configureStore(initStore);

  return render(ui, {
    wrapper: ({ children }) => (
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <ConnectedRouter history={history}>{children}</ConnectedRouter>
        </Provider>
      </ThemeProvider>
    ),
  });
}
