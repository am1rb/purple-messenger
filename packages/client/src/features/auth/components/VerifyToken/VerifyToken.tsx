import React, { memo, useEffect } from "react";
import { useDispatch } from "react-redux";
import { verifyToken } from "@purple-messenger/core";

export interface VerifyTokenProps {
  token: string;
}

function VerifyToken({ token }: VerifyTokenProps) {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(verifyToken(token));
  }, [dispatch, token]);

  return <>verify token...</>;
}

export default memo(VerifyToken);
