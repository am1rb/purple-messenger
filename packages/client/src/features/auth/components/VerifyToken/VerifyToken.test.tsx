import React from "react";
import { verifyToken } from "@purple-messenger/core";
import { renderWithStore } from "core/test";
import VerifyToken from "./VerifyToken";

describe("The <VerifyToken /> tests", () => {
  it("Should match the snapshot", () => {
    const { container } = renderWithStore(<VerifyToken token="" />);
    expect(container).toMatchSnapshot();
  });

  it("Should call verifyToken on mount event", () => {
    const token = "sample token";
    renderWithStore(<VerifyToken token={token} />);
    expect(verifyToken).toHaveBeenCalledWith(token);
  });
});
