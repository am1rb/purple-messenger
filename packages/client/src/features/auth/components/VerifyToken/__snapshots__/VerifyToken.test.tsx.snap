// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <VerifyToken /> tests Should match the snapshot 1`] = `
<div>
  verify token...
</div>
`;
